utils/docker/Dockerfile.checkpatch

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2021 Intel Corporation
+# All rights reserved.
+#
+# 'recipe' for Docker to build a image for running checkpatch.
+#
+
+# Pull base image
+FROM centos:centos7
+
+ARG CACHEBUST
```

Comment:
If you move this lower in the file, you will get more re-use at the higher layers.  I.e. if you do install first then cache-bust then upgrade, you will get longer life out of that pre-upgrade layer and less work in the lower layers.

Maybe we want to consider multiple layers of cache-busting.  One weekly and one daily (and/or one at the sub-hour value).  That way you are getting to-the-day (or hour or minute) updates in that final layer, but that operation should go quickly because you are leveraging a weekly layer above it.

Reply:
For checkpatch/codespell weekly is probably frequent enough, and the package list is short enough I didn't see much benefit in doing this multi-layered.  It really is a very small amount of work in this file now.

Reply:
For this effort that's probably fair enough.  I think I was making my comment more generically though, probably more applicable to `Dockerfile`s that build our software and then perhaps only apply that here in any effort to maintain a single structure for all of our Dockerfiles.

The other thing to consider in the frequency of updates is security and how long you want a given Docker image to be vulnerable to a particular security issue.

Reply:
That's a good point, in fact it sounds like this is demonstrated quite well with VM images in that they're updated every time but the base image are quite old.  I guess we want one level to update "frequently" to ensure the software is up-to-date and another level to force a complete rebuild "infrequently" to ensure the update doesn't start taking too long.

Reply:
> sounds like this is demonstrated quite well with VM images in that they're updated every time but the base image are quite old.

Yeah, we really have to get (and we are getting there -- just so much to do and so little resources to apply) to a point of having completely hands-off updating of the images on the CI test nodes.  Then we can update them at whatever frequency makes sense.  For IML, we used to do it once every 24h.  So one test run in a 24h time window would take a little bit longer than the rest on every cluster as the base was rebuilt.  I never really saw it as being all that advantageous (in respect to the effort to implement it) to make the image updating and replacing asynchronous so that even the 1 test run in the 24 time window didn't incur the extra build time.  But that could be done if it was really seen as an issue.